tasks/main.yml

```diff
@@ -38,5 +37,37 @@
     dest: /etc/sysconfig/kdump
     backup: yes
     mode: "{{ __sysconfig_kdump.stat.mode | d('0644') }}"
-  when: kexec_crash_size.content|b64decode|int > 0
   notify: restart kdump
+
+- name: Find out reserved memory for the crash kernel
+  slurp:
+    src: /sys/kernel/kexec_crash_size
+  register: kexec_crash_size
+
+- name: Set the kdump_reboot_required fact
+  set_fact:
+    kdump_reboot_required: "{{ kexec_crash_size.content|b64decode|int < 1 }}"
+
+- name: Fail if reboot is required and kdump_reboot_ok is false
+  fail:
+    msg: >-
+      "Reboot is required to apply changes. Re-execute the role after boot to
+      ensure that kdump is working."
+  when:
+    - kdump_reboot_required | bool
+    - not kdump_reboot_ok
```

Comment:
will this cause existing playbooks that use the kdump role to fail, that do not currently fail?

Reply:
The playbook will run smoothly on machines that run the kdump role previously, and therefore have kdump configured.
Replying to your quiestion, if you check CI from previous PRs, th role simply skipped all tasks on machines that did not have memory reserved for crash kernal. In other words, if machine contained `0` at `/sys/kernel/kexec_crash_size`, the role skipped all tasks and did not enable kdump. That was bad because users migh think that their kdump has been configured correctly, while in reality nothing worked.
I think this change makes the role work correctly.

Reply:
In short, the role fails if it really needs to fail. Previously, the role skipped all tasks and succeeded, keeping kdump not configured.